Conduit3 review comments

talbaneth opened the pull request.

talbaneth committed:
Conduit3 review comments

amusingaxl commented on src/conduits/RwaOutputConduit3.sol:
```diff
@@ -354,7 +354,7 @@ contract RwaOutputConduit3 {
      * @param wad The DAI amount.
      */
-    function _doQuit(uint256 wad) internal {
+    function _doQuit(uint256 wad) internal { // TB - as discussed I think the pattern of having internal functions right after their wrappers is nicer - https://github.com/makerdao/dss-vest/blob/master/src/DssVest.sol#L232
```
Since it's just a matter of preference, this is one of the coding standards that we established in CES.

amusingaxl commented on src/conduits/RwaOutputConduit3.sol:
```diff
@@ -364,7 +364,7 @@ contract RwaOutputConduit3 {
     //////////////////////////////////*/
 
-    uint256 internal constant WAD = 10**18;
+    uint256 internal constant WAD = 10**18; // TB - as discussed maker's convention is to put the MATH part in the beginning
```
Since it's just a matter of preference, this is one of the coding standards that we established in CES.

amusingaxl commented on src/conduits/RwaOutputConduit3.sol:
```diff
@@ -278,7 +278,7 @@ contract RwaOutputConduit3 {
      * @dev `msg.sender` must have been `mate`d and `to` must have been `pick`ed.
      */
-    function push() external isMate {
+    function push() external isMate { // TB - consider adding a slippage check here and in the other push function for the output, to protect against unexpected fee changes. I.e add a parameter of minGemAmount. It's pretty common for swapping logic.
```
Slippage protection seems like an overkill here, since the fees can only be changed by governance, with a very well defined schedule and this function will rarely be called. The chances of `push()` being "front-runned" with a spell execution are pretty slim.

talbaneth commented:
Hey @amusingaxl thanks for addressing these comments. This draft was my attempt of writing the comments somewhere.
However after we discussed I we'll actually finalize all our comments together, discuss them internally and then let you guys know when they're ready to be addressed.

amusingaxl commented on src/conduits/RwaInputConduit3.sol:
```diff
@@ -189,7 +189,7 @@ contract RwaInputConduit3 {
         } else if (what == "to") {
             require(data != address(0), "RwaInputConduit3/invalid-to-address");
-            to = data;
+            to = data; // TB - why is `to` fileable as opposed to RwaInputConduit2?
```
Input conduit `to` should be only the `RwaUrn` it's attached to.
In previous versions there not even a possibility to change it.

This address is shared with a crypto broker to allow them send payments back to MakerDAO. We learned that changing addresses is **very** cumbersome because everything is tied up in legal documents, so we added this escape hatch that can only be accessed by governance if we ever need to change it.

amusingaxl commented:
> Hey @amusingaxl thanks for addressing these comments. This draft was my attempt of writing the comments somewhere. However after we discussed it we'll actually finalize all our comments together, discuss them internally and then let you guys know when they're ready to be addressed.

I've created a PR containing only the changes made by @nduchak after our last call here: https://github.com/makerdao/mip21-toolkit/pull/20

Maybe it's more productive if you move your reviews there.

I'm gonna close this one.

talbaneth commented:
Yeap, we'll move everything there, thanks.
